web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/SingleUseKeyService.java

```diff
@@ -70,37 +78,67 @@ public boolean keyIsValid(String key) {
     }
 
     /**
-     * Invalidates a key by removing its entry from the CSV
-     * @param key access key of the box-c record
+     * Invalidates a single use key CSV record by removing it from the CSV
+     * @param key access key of the box-c record, may be null for time-based invalidation
      */
     public void invalidate(String key) {
         lock.lock();
         try {
             var csvRecords = parseCsv(CSV_HEADERS, csvPath);
             var updatedRecords = new ArrayList<>();
-            var keyExists = false;
+            var recordExists = false;
```

Comment:
I find the variable a bit confusing for what its used for (recordExists means we deleted a record), maybe `recordsRemoved` or `recordsChanged` might be clearer, particularly down in the conditional to decide whether to write out the CSV

Reply:
good point, thanks!